smile/src/main/java/tech/tablesaw/api/ml/clustering/LinkageFactory.java

```diff
@@ -0,0 +1,65 @@
+package tech.tablesaw.api.ml.clustering;
+
+import smile.clustering.linkage.CompleteLinkage;
+import smile.clustering.linkage.Linkage;
+import smile.clustering.linkage.SingleLinkage;
+import smile.clustering.linkage.UPGMALinkage;
+import smile.clustering.linkage.UPGMCLinkage;
+import smile.clustering.linkage.WPGMALinkage;
+import smile.clustering.linkage.WPGMCLinkage;
+import smile.clustering.linkage.WardLinkage;
+
+/**
+ * TODO
+ */
+public class LinkageFactory {
+	
+	
+	/**
+	 * TODO
+	 */
+	public enum type {
```

Comment:
Is there a better way to represent the different types of linkage methods than the one I have used here (i.e., placing a public enum inside a class whose job is to return classes, where the enum represents the different types of classes that can be created)?